merge/update.go

```diff
@@ -34,6 +36,14 @@ type Updater struct {
 	IgnoredFields map[fieldpath.APIVersion]*fieldpath.Set
 
 	enableUnions bool
+
+	// Stop comparing the new object with old object after applying.
+	// This was initially used to avoid spurious etcd update, but
+	// since that's vastly inefficient, we've come-up with a better
+	// way of doing that. Create this flag to stop it.
+	// Comparing has become more expensive too now that we're not using
+	// `Compare` but `value.Equals` so this gives an option to avoid it.
+	StopComparingAfterApply atomic.Bool
```

Comment:
why does this need to be atomic

Reply:
because it's not in the constructor so it seems like anyone can change it at anytime which is awful, so ... what do you suggest?

Reply:
Might've suggesting adding a way to put it into constructor but that wouldnt be api compatible. Was just curious